### DISCUSSION TITLE
barplot generation with gene-level adjusted p values

### DISCUSSION THREAD

**Discussion by IwanParf** [churchmanlab/genewalk#78] [Ideas]
Helloo,
It would be nice to be able to pre-define what bar plots should be generated with the default being global adjusted p values (the non-changeable default right now) which one could change with a flag to gene level adjusted p values (the right p value to look at if one is interested in one  or a few genes, right? btw I noticed that global adjusted p values are lower than gene-level adjusted p values which is odd in my opinion, since the global adjustment should be more strict?). Thank you in advance :)

**Comment by ri23** [comment]
Hi @IwanParf ,
Thank you for your interest in GeneWalk. Indeed, the default the bar plots are generated using global_padj. You can generate the gene level p-adj for individual genes using code described in the tutorial - section 
https://churchman.med.harvard.edu/genewalk/#tutorial Section 5. Visualization
Scripts in Python and R:
https://churchman.med.harvard.edu/s/GeneWalk_Python_Barplot_webpage_20210128.html
https://churchman.med.harvard.edu/s/GeneWalk_R_Barplot_webpage_20210128.html

Unfortunately, I don't have any bandwidth to build in a flag as you described. Thank you for your understanding.
It is possible that a global adjusted p-values are lower than gene-level, although I agree this is less intuitive. 

From https://en.wikipedia.org/wiki/False_discovery_rate
Benjamini Hochberg procedure is a method to find data with FDR < set significance level alpha
m= total number of tests
	1. For a given alpha , find the largest k such that P ( k ) ≤ k / m * alpha 
	2. Reject the null hypothesis (i.e., declare discoveries) for all H ( i ) for i = 1 , … , k
		( even for the ones that aren't less than their Benjamini-Hochberg critical value.)
For global level, the m is much larger than gene level, so more terms the k can vary between gene and global and lead to more terms being significant in global. If in doubt, I would interpret these results conservatively and focus on the GO terms cases where both global and gene levels p-adj are significant. I hope this clarifies.
Robert
